test/test_itasks_service.py

```diff
@@ -105,6 +141,134 @@ def test_process_data_incorrect_instance_no(self, mocked_popen):
+        method: start_server
+        state: Server started on Unix platform
+        expected_result: Server started
+        """
+        # Assign
+        itasksservice = ItasksService()
+        os.name = "posix"
+
+        # Act
+        itasksservice.start_server()
+
+        # Assert
+        mocked_popen.assert_called()
+        mocked_thread.assert_called()
+        self.assertTrue(mocked_thread.daemon)
+
+    def test_start_server_on_unsupported_system(self):
+        # pylint: disable=R0201, C0103
+        """
+        method: start_server
+        state: Server started on Mac platform
+        expected_result: Exception
+        """
+        # Assign
+        itasksservice = ItasksService()
+        os.name = "mac"
+
+        # Assert
+        self.assertRaises(UnsupportedOperatingSystemException,
+                          itasksservice.start_server)
+
+    def test_stop_server(self):  # pylint: disable=R0201, C0103
+        """
+        method: stop_server
+        expected_result: stop method called on process
+        """
+        # Assign
+        itasksservice = ItasksService()
+        itasksservice.process = Mock()
+        itasksservice.process.pid = 999999
+        os.kill = Mock()
+
+        # Act
+        itasksservice.stop_server()
+
+        # Assert
+        itasksservice.process.kill.assert_called()
+        os.kill.assert_called_with(999999, 2)
+
+    def test_non_block_read(self):
+        """
+        method: non_block_read
+        state: Line is available longer than 1 character
+        expected_result: String is returned
+        """
+        # Assign
+        itasksservice = ItasksService()
+        method_input = Mock()
+        method_input.readline = Mock()
+        method_input.readline.return_value = "Test string ".encode()
+        expected_result = "Test string"
+
+        # Act
+        result = itasksservice.non_block_read(method_input)
+
+        # Assert
+        self.assertEqual(result, expected_result)
+
+    def test_non_block_read_nothing_to_read(self):
+        # pylint: disable=C0103
+        """
+        method: non_block_read
+        state: No data available
+        expected_result: String is returned
+        """
+        # Assign
+        itasksservice = ItasksService()
+        method_input = Mock()
+        method_input.readline = Mock()
+        method_input.readline.return_value = "".encode()
+        expected_result = None
+
+        # Act
+        result = itasksservice.non_block_read(method_input)
+
+        # Assert
+        self.assertEqual(result, expected_result)
+
+    def test_non_block_read_failed(self):
+        """
+        method: non_block_read
+        state: Exception is thrown
+        expected_result: Exception is thrown
+        """
+        # Assign
+        itasksservice = ItasksService()
+        method_input = Mock()
+        method_input.readline = Mock(side_effect=Exception)
+
+        # Assert
+        self.assertRaises(CouldNotReadStdIOException,
```

Comment:
Usually done with 'with self.assertRaises'  on the act. Since you act and assign in the same place.

Reply:
Same for above